### ISSUE TITLE
Screen Crashing Game When Pressing Tab

### ISSUE THREAD

**Issue by Kvarts1** [McJtyMods/RFToolsUtility#120]
Hello, I'm playing on a server with a modpack(https://www.curseforge.com/minecraft/modpacks/tnp-limitless-5).

The Issue:
My game crashes whenever I open the playerlist or tab list after I place down a Screen and if I break the Screen I can open the tab menu again without crashing.
I can still press tab without crashing if i unbind the list players keybind.
[latestEdited.log](https://github.com/McJtyMods/RFToolsUtility/files/7870666/latestEdited.log)
(The Edit is to remove my name)

**Comment by McJty** [comment]
I will check it out. Thanks for reporting

**Comment by McJty** [comment]
I cannot reproduce this and it's also not crashing in any of my code so not sure what to do about this

**Comment by craftish37** [comment]
Hello, just to say that the block who's making the game crash when pressing tab is the Environmental Controller 👀

**Comment by McJty** [comment]
You are not using the last version of my mods. Try updating

**Comment by lyuma** [comment]
After updating, AllTheMods7 still giving a crash with RfToolsUtility 1.18-4.0.10 when hitting tab:

```
FATAL ERROR in native method: Thread[Render thread,10,main]: No context is current or a function that is not available in the current context was called. The JVM will abort execution.
    at org.lwjgl.opengl.GL11.glColor4f(org.lwjgl.opengl/Native Method)
    at mcjty.rftoolsutility.client.RenderGameOverlayEventHandler.lambda$renderBuffs$0(rftoolsutility@1.18-4.0.10/RenderGameOverlayEventHandler.java:70)
    at mcjty.rftoolsutility.client.RenderGameOverlayEventHandler$$Lambda$37692/0x00000008043f5810.accept(rftoolsutility@1.18-4.0.10/Unknown Source)
    at net.minecraftforge.common.util.LazyOptional.ifPresent(forge@39.0.88/LazyOptional.java:164)
    at mcjty.rftoolsutility.client.RenderGameOverlayEventHandler.renderBuffs(rftoolsutility@1.18-4.0.10/RenderGameOverlayEventHandler.java:43)
    at mcjty.rftoolsutility.client.RenderGameOverlayEventHandler.onRender(rftoolsutility@1.18-4.0.10/RenderGameOverlayEventHandler.java:30)
    at mcjty.rftoolsutility.setup.ClientSetup.renderGameOverlayEvent(rftoolsutility@1.18-4.0.10/ClientSetup.java:33)
    at mcjty.rftoolsutility.RFToolsUtility$$Lambda$17565/0x00000008025bb660.accept(rftoolsutility@1.18-4.0.10/Unknown Source)
```

The crash occurs when rendering a powered Environmental Controls block while a buff module (e.g. Regeneration) is installed within, and while holding tab (player list).

If the block is out of range or not powered, it will not crash.

My understanding is immediate mode OpenGL is deprecated or no longer supported when using a modern OpenGL, and indeed this is the only place in the code where operations such as glColor4f or GL_LIGHTING (which is incompatible with shaders) are used.
```
            GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
            GL11.glDisable(GL11.GL_LIGHTING);
```
I would suggest removing these two lines, as nothing good will come of them.

**Comment by McJty** [comment]
Will be fixed next version (1.18.2 only though)
